pkg/api/componentreadiness/test_details.go

```diff
@@ -50,60 +56,74 @@ func (c *ComponentReportGenerator) GenerateTestDetailsReport(ctx context.Context
 		}
 	}
 
-	componentJobRunTestReportStatus, errs := c.GenerateJobRunTestReportStatus(ctx)
+	before := time.Now()
+
+	// load all pass/fails for specific jobs, both sample, basis, and override basis if requested
+	componentJobRunTestReportStatus, errs := c.getJobRunTestStatusFromBigQuery(ctx)
 	if len(errs) > 0 {
 		return crtype.ReportTestDetails{}, errs
 	}
+
+	logrus.Infof("getJobRunTestStatusFromBigQuery completed in %s with %d sample results and %d base results from db", time.Since(before), len(componentJobRunTestReportStatus.SampleStatus), len(componentJobRunTestReportStatus.BaseStatus))
+	now := time.Now()
+	componentJobRunTestReportStatus.GeneratedAt = &now
+
+	// Allow all middleware a chance to transform the job run test statuses:
 	var err error
+	for _, mw := range c.middlewares {
+		err = mw.TransformTestDetails(&componentJobRunTestReportStatus)
+		if err != nil {
+			return crtype.ReportTestDetails{}, []error{err}
+		}
+	}
+
 	bqs := NewBigQueryRegressionStore(c.client)
 	allRegressions, err := bqs.ListCurrentRegressions(ctx)
 	if err != nil {
 		errs = append(errs, err)
 		return crtype.ReportTestDetails{}, errs
 	}
 
-	var baseOverrideReport *crtype.ReportTestDetails
-	if c.ReqOptions.BaseOverrideRelease.Release != "" && c.ReqOptions.BaseOverrideRelease.Release != c.ReqOptions.BaseRelease.Release {
-		// because internalGenerateTestDetailsReport modifies SampleStatus we need to copy it here
-		overrideSampleStatus := map[string][]crtype.JobRunTestStatusRow{}
-		for k, v := range componentJobRunTestReportStatus.SampleStatus {
-			overrideSampleStatus[k] = v
+	// Generate the report for the main release that was originally requested:
+	c.openRegressions = FilterRegressionsForRelease(allRegressions, c.ReqOptions.SampleRelease.Release)
+	report := c.internalGenerateTestDetailsReport(ctx,
+		componentJobRunTestReportStatus.BaseStatus,
+		c.ReqOptions.BaseRelease.Release,
+		&c.ReqOptions.BaseRelease.Start,
+		&c.ReqOptions.BaseRelease.End,
+		componentJobRunTestReportStatus.SampleStatus)
+	report.GeneratedAt = componentJobRunTestReportStatus.GeneratedAt
+
+	for _, mw := range c.middlewares {
+		err = mw.TestDetailsAnalyze(&report)
+		if err != nil {
+			return crtype.ReportTestDetails{}, []error{err}
 		}
+	}
 
-		overrideReport := c.internalGenerateTestDetailsReport(ctx, componentJobRunTestReportStatus.BaseOverrideStatus, c.ReqOptions.BaseOverrideRelease.Release, &c.ReqOptions.BaseOverrideRelease.Start, &c.ReqOptions.BaseOverrideRelease.End, overrideSampleStatus)
+	// Generate the report for the fallback release if one was found:
+	// Move all this to the middleware.
```

Comment:
This is poised to move but the use of c.internalGenerateTestDetailsReport gets very complicated particularly with triage. I need to figure out how to extract that before I can proceed here, but picture this block of code in the middleware.

Reply:
Technically I don't think we need `var baseOverrideReport *crtype.ReportTestDetails` any longer, you could just append `overrideReport`.  But with your comment that there is further refactoring to be done seems fine to leave as is for now.

Reply:
I will note that for the next phase, ty!